Adds `super` as a named modifier in lf key expressions.

diff --git a/key.go b/key.go
--- a/key.go
+++ b/key.go
@@ -135,8 +135,8 @@ func init() {
 	}
 }
 
-// for simplicity, assume there will only be one modifier (ctrl, shift or alt)
-var reModKey = regexp.MustCompile(`<(c|s|a)-(.+)>`)
+// for simplicity, assume there will only be one modifier
+var reModKey = regexp.MustCompile(`<(c|s|a|super)-(.+)>`)
 
 func wrapModifier(s string, mod string) string {
 	s = strings.TrimPrefix(s, "<")
@@ -156,6 +156,8 @@ func addKeyModifier(s string, mod tcell.ModMask) string {
 		return wrapModifier(s, "s")
 	case mod&tcell.ModAlt != 0:
 		return wrapModifier(s, "a")
+	case mod&tcell.ModMeta != 0:
+		return wrapModifier(s, "super")
 	default:
 		return s
 	}
@@ -195,6 +197,8 @@ func parseKeyModifier(s string) (tcell.ModMask, string) {
 		mod = tcell.ModShift
 	case "a":
 		mod = tcell.ModAlt
+	case "super":
+		mod = tcell.ModMeta
 	}
 
 	s = matches[2]
diff --git a/key_test.go b/key_test.go
--- a/key_test.go
+++ b/key_test.go
@@ -17,10 +17,12 @@ var gKeyTests = []struct {
 	{tcell.NewEventKey(tcell.KeyCtrlA, "", tcell.ModNone), "<c-a>"},
 	{tcell.NewEventKey(tcell.KeyRune, "A", tcell.ModNone), "A"},
 	{tcell.NewEventKey(tcell.KeyRune, "a", tcell.ModAlt), "<a-a>"},
+	{tcell.NewEventKey(tcell.KeyRune, "a", tcell.ModMeta), "<super-a>"},
 	{tcell.NewEventKey(tcell.KeyLeft, "", tcell.ModNone), "<left>"},
 	{tcell.NewEventKey(tcell.KeyLeft, "", tcell.ModCtrl), "<c-left>"},
 	{tcell.NewEventKey(tcell.KeyLeft, "", tcell.ModShift), "<s-left>"},
 	{tcell.NewEventKey(tcell.KeyLeft, "", tcell.ModAlt), "<a-left>"},
+	{tcell.NewEventKey(tcell.KeyLeft, "", tcell.ModMeta), "<super-left>"},
 	{tcell.NewEventKey(tcell.KeyEsc, "", tcell.ModNone), "<esc>"},
 	{tcell.NewEventKey(tcell.KeyF1, "", tcell.ModNone), "<f-1>"},
 }
